kratos/containers/variables_list_data_value_container.h

```diff
@@ -325,32 +325,30 @@ class KRATOS_API(KRATOS_CORE) VariablesListDataValueContainer
     template<class TDataType>
     TDataType& GetValue(const Variable<TDataType>& rThisVariable)
     {
-        if(!mpVariablesList->Has(rThisVariable))
-            KRATOS_THROW_ERROR(std::invalid_argument, "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:",rThisVariable);
+        KRATOS_ERROR_IF_NOT(mpVariablesList->Has(rThisVariable)) << "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:" << rThisVariable << std::endl;
         return *(TDataType*)Position(rThisVariable);
     }
 
     template<class TDataType>
     TDataType& GetValue(const Variable<TDataType>& rThisVariable, SizeType QueueIndex)
     {
-        if(!mpVariablesList->Has(rThisVariable))
-            KRATOS_THROW_ERROR(std::invalid_argument, "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:",rThisVariable);
+        KRATOS_ERROR_IF_NOT(mpVariablesList->Has(rThisVariable)) << "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:" << rThisVariable << std::endl;
+        KRATOS_DEBUG_ERROR_IF((QueueIndex + 1) > mQueueSize) << "ERROR:: YOUR ARE TRYING TO ACCES A VALUE OUTSIDE THE BUFFER SIZE: " << mQueueSize << std::endl;
```

Comment:
It would be better if you tell the user with buffer step is calling which is out of bound. And as @philbucher mentions, the `ERROR::` in the message should be added by logger output not in yours.